fvm/src/engine/mod.rs

```diff
@@ -491,7 +491,7 @@ impl Engine {
         let module = match cache.get(k) {
             Some(m) => m.module.clone(),
             None => {
-                let module = Module::deserialize(&self.inner.engine, compiled)?;
+                let module = unsafe { Module::deserialize(&self.inner.engine, compiled)? };
```

Comment:
what's going on here? Did wasmtime mark this as unsafe internally and now 2024 is forcing us to acknowledge it?

Reply:
The method we're calling this from is marked as `unsafe`. Previously, functions marked as unsafe could call unsafe functions without declaring an unsafe block. Now, marking the function as unsafe doesn't let you call unsafe functions without declaring a separate unsafe block.